File: RT_Inventory.Api/RT_Inventory.Api/Models/Entities/Requisition.cs
using RT_Inventory.Api.Models.Enums;

namespace RT_Inventory.Api.Models.Entities;

public class Requisition
{
    public int Id { get; set; }
    public string RequisitionNo { get; set; } = string.Empty;
    public RequisitionPurpose Purpose { get; set; }
    public RequisitionStatus Status { get; set; } = RequisitionStatus.Pending;
    public int CreatedByUserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public User CreatedByUser { get; set; } = null!;
    public ICollection<RequisitionItem> Items { get; set; } = [];
    public ICollection<PickingList> PickingLists { get; set; } = [];
}

File: RT_IMS_front-end/rt_ims_front-end/src/types/requisition.ts
import type { EntityId } from "./auth";

export type RequisitionPurpose = "Knitting" | "YD" | "Testing";
export type RequisitionStatus = "Draft" | "Pending" | "Picked" | "Picking" | "Issued" | "Closed" | "Cancelled";

export interface RequisitionItem {
  id: EntityId;
  itemYarnType: string;
  lotNo: string;
  bagQty: number;
  requiredBagQty?: number;
  requiredWeight: number;
}

export interface PickingList {
  id: EntityId;
  requisitionId: EntityId;
  pickingListNo: string;
  barcodeValue: string;
  createdByUsername?: string;
  createdAt?: string;
}

export interface Requisition {
  id: EntityId;
  requisitionNo: string;
  purpose: RequisitionPurpose;
  status: RequisitionStatus | string;
  createdBy: string;
  date: string;
  items: RequisitionItem[];
  pickingLists?: PickingList[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateRequisitionPayload {
  purpose: RequisitionPurpose;
  items: Array<{
    itemYarnType: string;
    lotNo?: string;
    requiredBagQty: number;
    requiredWeight: number;
  }>;
}
